stubs/assertpy/assertpy/dynamic.pyi

```diff
@@ -0,0 +1,10 @@
+from collections.abc import Iterable as _Iterable
+from typing import Any
+from typing_extensions import Self, TypeAlias
+
+__tracebackhide__: bool
+
+Iterable: TypeAlias = _Iterable[Any]
+
+class DynamicMixin:
+    def __getattr__(self, attr: str) -> Self: ...
```

Comment:
This is a bit unfortunate because if you use a typo on an attribute, you won't get an immediate error. Looking at the implementation, it only supports attributes starting with `has_`, but that can't be expressed in the type system.

However, it returns a callable, not a Self:
```suggestion
    def __getattr__(self, attr: str) -> Callable[..., Self]: ...
```

Reply:
Oh, I missed that, fixed!